plugins/importexport/users/filter/UserXmlPKPUserFilter.inc.php

```diff
@@ -105,24 +108,33 @@ function parseUser($node) {
 			case 'auth_string': $user->setAuthString($n->textContent); break;
 			case 'disabled_reason': $user->setDisabledReason($n->textContent); break;
 			case 'locales': $user->setLocales(preg_split('/:/', $n->textContent)); break;
-
 			case 'password':
 				if ($n->getAttribute('must_change') == 'true') {
 					$user->setMustChangePassword(true);
 				}
+
 				if ($n->getAttribute('is_disabled') == 'true') {
 					$user->setIsDisabled(true);
 				}
+
+				if ($n->getAttribute('encryption')) {
+					$encryption = $n->getAttribute('encryption');
+				}
+
 				$passwordValueNodeList = $n->getElementsByTagNameNS($deployment->getNamespace(), 'value');
 				if ($passwordValueNodeList->length == 1) {
 					$password = $passwordValueNodeList->item(0);
 					$user->setPassword($password->textContent);
 				} else {
 					fatalError("User has no password.  Check your import XML format.");
 				}
+
 				break;
 		}
 
+		// Password Import Validation
+		$password = Validation::importUserPasswordValidation($user, $encryption);
```

Comment:
This function is only used here, and eventually I'd like to remove the `Validation` class (or at least keep it as small as possible). I'd suggest having the contents of that function live in this class instead.

Reply:
Done.